src/services/state-persistor/utils/FeatureStyleHelper.ts

```diff
@@ -0,0 +1,911 @@
+import Feature from 'ol/Feature.js'
+import Map from 'ol/Map.js'
+import { Coordinate } from 'ol/coordinate.js'
+import ngeoFormatFeatureProperties from './FeatureProperties'
+import ngeoGeometryType from './GeometryTypes'
+import * as olArray from 'ol/array.js'
+import * as olColor from 'ol/color.js'
+import * as olExtent from 'ol/extent.js'
+import olGeomLineString from 'ol/geom/LineString.js'
+import olGeomMultiLineString from 'ol/geom/MultiLineString.js'
+import olGeomMultiPoint from 'ol/geom/MultiPoint.js'
+import olGeomPoint from 'ol/geom/Point.js'
+import olGeomPolygon from 'ol/geom/Polygon.js'
+import olGeomMultiPolygon from 'ol/geom/MultiPolygon.js'
+import olStyleCircle from 'ol/style/Circle.js'
+import olStyleFill from 'ol/style/Fill.js'
+import olStyleRegularShape from 'ol/style/RegularShape.js'
+import olStyleStroke from 'ol/style/Stroke.js'
+import olStyleStyle from 'ol/style/Style.js'
+import olStyleText from 'ol/style/Text.js'
+import * as olProj from 'ol/proj.js'
+import {
+  getFormattedLength,
+  getFormattedArea,
+  getFormattedAzimutRadius,
+  getFormattedPoint,
+} from '@/services/common/measurement.utils'
+
+class FeatureStyleHelper {
+  private precision_: number | undefined = undefined
+  private projection_: olProj.Projection | undefined
+  private decimals_ = 3
+  private pointFilterFn_: Function | null = null
```

Comment:
can we provide a more precise signature than `Function`?

Reply:
it's a one shot port from v3 with limited efforts on type safety